File: frontend/lib/data/Judge.dart
import 'dart:io';

import 'package:db_finalproject/data/User.dart';

class Judge extends User{
  String? title;

  Judge({
    required super.id,
    super.passwd,
    required super.name,
    required super.email,
    required super.sexual,
    required super.phone,
    this.title
  });

  factory Judge.fromJson(Map<String, dynamic> json){
    return Judge(
      id: json['judgeid'] as String,
      name: json['judgename'] as String, 
      email: json['judgeemail'] as String, 
      sexual: json['judgesexual'] as String, 
      phone: json['judgephone'] as String,
      title: json['judgetitle'] as String?
    );
  }

  @override
  Map<String,dynamic> toJson(){
    return {
      'judgeid':email,
      'judgepasswd': passwd,
      'judgename':name,
      'judgesexual':sexual,
      'judgephone':phone,
      'judgetitle':title
    };
  }

  @override
  void setField(String field,String value){
    switch (field) {
      case 'judgeid':
        id = value;
        break;
      case 'judgepasswd':
        passwd = value;
        break;
      case 'judgename':
        name = value;
        break;
      case 'judgeemail':
        email = value;
        break;
      case 'judgesexual':
        sexual = value;
        break;
      case 'judgephone':
        phone = value;
        break;
      case 'judgetitle':
        title = value;
        break;
      default:
        break;
    }
  }

  @override
  String getField(String field){
    switch (field) {
      case 'judgeid':
        return id;
      case 'judgepasswd':
        return passwd!;
      case 'judgename':
        return name;
      case 'judgeemail':
        return email;
      case 'judgesexual':
        return sexual;
      case 'judgephone':
        return phone;
      case 'judgetitle':
        return title!;
      default:
        return '';
    }
  }
}

File: backend/src/main/java/com/example/demo/model/Judge.java
package com.example.demo.model;

import com.fasterxml.jackson.annotation.JsonProperty;

// 'judgeid':email,
//       'judgepasswd': passwd,
//       'judgename':name,
//       'judgesexual':sexual,
//       'judgephone':phone,
//       'judgetitle':title
public class Judge {
    @JsonProperty("judgeid")
    private String judgeid;
    @JsonProperty("judgepasswd")
    private String judgepasswd;
    @JsonProperty("judgename")
    private String judgename;
    @JsonProperty("judgesexual")
    private String judgesexual;
    @JsonProperty("judgephone")
    private String judgephone;
    @JsonProperty("judgetitle")
    private String judgetitle;
    @JsonProperty("judgeemail")
    private String judgeemail;

    public void setJudgeid(String judgeid) {
        this.judgeid = judgeid;
    }

    public void setJudgepasswd(String judgepasswd) {
        this.judgepasswd = judgepasswd;
    }

    public void setJudgename(String judgename) {
        this.judgename = judgename;
    }

    public void setJudgesexual(String judgesexual) {
        this.judgesexual = judgesexual;
    }

    public void setJudgephone(String judgephone) {
        this.judgephone = judgephone;
    }

    public void setJudgetitle(String judgetitle) {
        this.judgetitle = judgetitle;
    }

    public String getJudgeid() {
        return judgeid;
    }

    public String getJudgepasswd() {
        return judgepasswd;
    }

    public String getJudgename() {
        return judgename;
    }

    public String getJudgesexual() {
        return judgesexual;
    }

    public String getJudgephone() {
        return judgephone;
    }

    public String getJudgetitle() {
        return judgetitle;
    }

    public String getJudgeemail() {
        return judgeemail;
    }

    public void setJudgeemail(String judgeemail) {
        this.judgeemail = judgeemail;
    }

    @Override
    public String toString() {
        return "Judge{" +
                "judgeid='" + judgeid + '\'' +
                ", judgepasswd='" + judgepasswd + '\'' +
                ", judgename='" + judgename + '\'' +
                ", judgesexual='" + judgesexual + '\'' +
                ", judgephone='" + judgephone + '\'' +
                ", judgetitle='" + judgetitle + '\'' +
                ", judgeemail='" + judgeemail + '\'' +
                '}';
    }
}
